source/resampling_with_code2.Rmd

```diff
@@ -0,0 +1,669 @@
+it.
+:::
+
+Imagine a factory robot that slides along a worktable on which several items
+have been placed.  It services each of these items, one by one, until it is
+done.
+
+So the robot knows *which* item it is operating on, each item is given
+a label.
+
+Here is an example:
+
+```{python, opts.label="py_ed"}
+for item in ['shoe', 'saddle', 'jacket']:
+    print('I am stitching a', item)
+```
+
+Here, the robot is working its way through three items: a shoe, a saddle, and a jacket.
+As it moves on to each object, it assigns the label ("shoe", "saddle", etc.) to the variable `item`.
+The indented block of code under the for is executed for each item, and in this code block `label` can be accessed.
+The code above can therefore be unpacked into the following equivalent program:
+
+```{python, opts.label="py_ed"}
+item = 'shoe'
+print('I am stitching a', item)
+
+item = 'saddle'
+print('I am stitching a', item)
+
+item = 'jacket'
+print('I am stitching a', item)
+```
+
+We used the name `item` here, but any other name would do.
+Thus, the following is equivalent:
+
+```{python, opts.label="py_ed"}
+for package in ['shoe', 'saddle', 'jacket']:
+    print('I am stitching a', package)
+```
+
+It is common to use a number as a label in a for loop:
+
+```{python, opts.label="py_ed"}
+for i in [0, 1, 2, 3, 4]:
+    print('The square of', i, 'is', i * i)
+```
+
+When we want to count to a hundred, it becomes laborious to type out the
+numbers. For that, Python has a shorthand called `range`:
+
+```{python, opts.label="py_ed"}
+for i in range(5):
+    print('The square of', i, ' is ', i * i)
+```
+:::
+
+## Putting it all together
+
+We found that we could construct the code for a single trial.  Here is the code for a single trial, again:
+
+```{python, opts.label="py_ed"}
+rnd = np.random.default_rng()
+# Get 12 random numbers from 0 through 99
+zero_through_99 = np.arange(0, 100)
+a = rnd.choice(zero_through_99, 12)
+# How many numbers are greater than 74?
+b = np.sum(a > 74)
+# Show the result
+b
+```
+
+```{r, opts.label="r_ed"}
+# Get 12 random numbers from 0 through 99
+a <- sample(0:99, 12, replace=TRUE)
+# How many are greater than 74?
+b <- sum(a == 9)
+# Show the result
+b
+```
+
+Then we found that we could use {{< var array_name >}}s to store the results
+of these trials, and that we could use `for` loops to do many repeats of steps
+like those for a single trial.
+
+Now we can put these parts together to do 100 simulated trials:
+
+```{python, opts.label="py_ed"}
+# Procedure for 100 simulated trials.
+
+# An array to store the counts for each trial.
+z = np.zeros(100)
+
+# Repeat the trial procedure 100 times.
+for i in np.arange(0, 100):
+    rnd = np.random.default_rng()
+    # Get 12 random numbers from 0 through 99
+    zero_through_99 = np.arange(0, 100)
+    a = rnd.choice(zero_through_99, 12)
+    # How many numbers are greater than 74?
+    b = np.sum(a > 74)
```

Comment:
Unnecessary; we can simply store `np.all(a > 74)`

Reply:
Hmm - yes - but the `np.all` is a special case, and we haven't covered 'np.all` yet.